server/fishtest/rundb.py

```diff
@@ -237,13 +237,9 @@ def new_run(
 
         return self.runs.insert_one(new_run).inserted_id
 
-    def get_machines(self):
+    def get_machines(self, unfinished_runs):
         machines = []
-        active_runs = self.runs.find(
-            {"finished": False, "tasks": {"$elemMatch": {"active": True}}},
-            sort=[("last_updated", DESCENDING)],
-        )
-        for run in active_runs:
+        for run in unfinished_runs:
```

Comment:
missing filtering by active?

Reply:
yes Indeed. but this was xoto idea here https://github.com/glinscott/fishtest/pull/1597/
Which I don't like.

We reduced 3 dp calls by this.
We can afford filtering in code here.

Reply:
the query should definitely stay, was thinking about filtering in the code, as I wasn't sure it's equivalent without it

Reply:
No queries in this line, we have the data passed, so I definitely agree with you that, to make this equivalent, A filtering in code to active runs would be best and not at all expensive

Reply:
There is already filtering in code right (line 244)?